src/plan/tracing.rs

```diff
@@ -1,104 +1,111 @@
 //! This module contains code useful for tracing,
 //! i.e. visiting the reachable objects by traversing all or part of an object graph.
 
-use std::mem;
-
 use crate::scheduler::gc_work::{EdgeOf, ProcessEdgesWork};
 use crate::scheduler::{GCWorker, WorkBucketStage};
 use crate::util::ObjectReference;
 use crate::vm::EdgeVisitor;
 
 /// This trait represents an object queue to enqueue objects during tracing.
-pub trait ObjectQueue {
+pub trait Queue<T> {
     /// Enqueue an object into the queue.
-    fn enqueue(&mut self, object: ObjectReference);
+    /// Returns `true` if the queue reaches it's capacity.
```

Comment:
There is no return value on the line below.

Reply:
Fixed